### PR TITLE
Move magic strings for KeyAction, TaskInfoStatus and TaskInfoType to …

### LINKED ISSUE #303: Use strong types to map lists instead of pure strings
The idea here is to implement `enum`'s to map lists eg:

- src/Meilisearch/Key.cs: `public IEnumerable<string> Actions { get; set; }`
- src/Meilisearch/QueryParamaters/TasksQuery.cs: `public List<string> Types { get; set; }` and `public List<string> Status { get; set; }`
- and others...

We may want to refactor this `src/Meilisearch/TaskInfo.cs`: `TaskInfoType`, `TaskInfoStatus` to a `static class` with public `const` fields.

### PR DESCRIPTION
**Pull request by juchom**
…enums

# Pull Request

## What does this PR do?
Fixes #303

KeyAction, TaskInfoStatus, TaskInfoType are enums.

A custom converter has been made for KeyAction to convert from PascalCase to Dot-Case.

## PR checklist
Please check if your PR fulfills the following requirements:
- [X] Does this PR fix an existing issue?
- [X] Have you read the contributing guidelines?
- [X] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Review by alallema — DISMISSED** [review]
Thanks @juchom for this PR ❤️ 
LGTM! I especially like the rewriting of the `TasksQuery` parameter definitions 😉

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -479,8 +479,8 @@ get_all_tasks_filtering_1: |-
   ResourceResults<TaskInfo> taskResult = await client.GetTasksAsync(new TasksQuery { IndexUid = "movies" });
 get_all_tasks_filtering_2: |-
   ResourceResults<TaskInfo> taskResult = await client.GetTasksAsync(new TasksQuery {
-    Status = new List<string>() { "succeeded", "failed" },
-    Types = new List<string>() { "documentAdditionOrUpdate" },
+    Status = new List<TaskInfoStatus>() { TaskInfoStatus.Succeeded, TaskInfoStatus.failed },
+    Types = new List<TaskInfoType>() { TaskInfoType.documentAdditionOrUpdate },
```
Comment (brunoocasali) [inline .code-samples.meilisearch.yaml]: It shouldn't be `TaskInfoType.DocumentAdditionOrUpdate`?

File: .code-samples.meilisearch.yaml [hunk 2]
```diff
@@ -479,8 +479,8 @@ get_all_tasks_filtering_1: |-
   ResourceResults<TaskInfo> taskResult = await client.GetTasksAsync(new TasksQuery { IndexUid = "movies" });
 get_all_tasks_filtering_2: |-
   ResourceResults<TaskInfo> taskResult = await client.GetTasksAsync(new TasksQuery {
-    Status = new List<string>() { "succeeded", "failed" },
-    Types = new List<string>() { "documentAdditionOrUpdate" },
+    Status = new List<TaskInfoStatus>() { TaskInfoStatus.Succeeded, TaskInfoStatus.failed },
```
Comment (brunoocasali) [inline .code-samples.meilisearch.yaml]: And `TaskInfoStatus.Succeeded, TaskInfoStatus.Failed`

**Review by brunoocasali — APPROVED** [review]
Awesome 🤩

**Comment by alallema** [comment]
bors merge

**Pull request merged**